src/components/KTableView/KTableView.vue

```diff
@@ -407,180 +402,47 @@ enum TableViewHeaderKeys {
-   * Column visibility/width.
-   */
-  tablePreferences: {
-    type: Object as PropType<TablePreferences>,
-    default: () => ({}),
-  },
-  /**
-   * Enables hover highlighting to table rows
-   */
-  rowHover: {
-    type: Boolean,
-    default: true,
-  },
-  /**
-   * A function that conditionally specifies row attributes on each row
-   */
-  rowAttrs: {
-    type: Function as PropType<(row: Record<string, any>) => Record<string, string>>,
-    default: () => ({}),
-  },
-  /**
-   * A function that conditionally turns a row into a link
-   */
-  rowLink: {
-    type: Function as PropType<(row: Record<string, any>) => RowLink>,
-    default: () => ({}),
-  },
-  /**
-   * A function that conditionally specifies whether bulk actions are disabled for a row and the tooltip to display. Default value: () => true
-   */
-  rowBulkActionEnabled: {
-    type: Function as PropType<(row: Record<string, any>) => RowBulkAction>,
-    default: () => true,
-  },
-  /**
-   * A function that conditionally specifies cell attributes
-   */
-  cellAttrs: {
-    type: Function,
-    default: () => ({}),
-  },
-  /**
-   * A prop that enables a loading skeleton
-   */
-  loading: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * A prop to pass in a custom empty state title
-   */
-  emptyStateTitle: {
-    type: String,
-    default: 'No Data',
-  },
-  /**
-   * A prop to pass in a custom empty state message
-   */
-  emptyStateMessage: {
-    type: String,
-    default: 'There is no data to display.',
-  },
-  /**
-   * A prop to pass in a custom empty state action route
-   */
-  emptyStateActionRoute: {
-    type: [Object, String] as PropType<RouteLocationRaw | string>,
-    default: null,
-  },
-  /**
-   * A prop to pass in a custom empty state action message
-   */
-  emptyStateActionMessage: {
-    type: String,
-    default: '',
-  },
-  emptyStateIconVariant: {
-    type: String as PropType<EmptyStateIconVariant>,
-    default: EmptyStateIconVariants.Default,
-  },
-  emptyStateButtonAppearance: {
-    type: String as PropType<ButtonAppearance>,
-    default: 'primary',
-  },
-  /**
-   * A prop that enables the error state
-   */
-  error: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * A prop to pass in a custom error state title
-   */
-  errorStateTitle: {
-    type: String,
-    default: 'An error occurred',
-  },
-  /**
-   * A prop to pass in a custom error state message
-   */
-  errorStateMessage: {
-    type: String,
-    default: 'Data cannot be displayed due to an error.',
-  },
-  /**
-   * A prop to pass in a custom error state action route
-   */
-  errorStateActionRoute: {
-    type: [Object, String] as PropType<RouteLocationRaw | string>,
-    default: null,
-  },
-  /**
-   * A prop to pass in a custom error state action message
-   */
-  errorStateActionMessage: {
-    type: String,
-    default: '',
-  },
+interface TableViewProps extends TableProps {
```

Comment:
nit: can we move this to the types file?

Reply:
(for both comments like this, my reasoning is that if they are defined here, they are not exported for import by the consuming app)

Reply:
> (for both comments like this, my reasoning is that if they are defined here, they are not exported for import by the consuming app)

Oh maybe this is why I can't see the inline comments in the consuming app. But then the shared interface doesn't have to be exported?

Reply:
> But then the shared interface doesn't have to be exported?

If the actual table prop interfaces are exported, the "TablePropsShared" does not need to be exported on it's own

```ts
interface TablePropsShared { ... }

export interface TableViewProps extends TablePropsShared & { ... }
export interface TableDataProps extends TablePropsShared & { ... }
```

Reply:
Done